pkg/operator/ceph/cluster/controller.go

```diff
@@ -143,8 +144,15 @@ func watchOwnedCoreObject[T client.Object](c controller.Controller, mgr manager.
 }
 
 func add(opManagerContext context.Context, mgr manager.Manager, r reconcile.Reconciler, context *clusterd.Context, opConfig opcontroller.OperatorConfig) error {
+	concurrentClusters := os.Getenv("ROOK_RECONCILE_CONCURRENT_CLUSTERS")
+	concurrentReconciles := 1
+	if i, err := strconv.Atoi(concurrentClusters); err == nil && i > 1 {
+		logger.Infof("Setting concurrent cluster reconciles to %d", i)
+		concurrentReconciles = i
+	}
+
 	// Create a new controller
-	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r, MaxConcurrentReconciles: concurrentReconciles})
```

Comment:
what will happen if the max is set more than the cephcluster crs? does it allow to set any max value irrespective of the crs

Reply:
It doesn't matter if the value is set higher than the number of CRs. You can only reconcile that count in parallel if you have that many CRs.